pkg/kanctl/repositoryserver.go

```diff
@@ -0,0 +1,233 @@
+	cmd.PersistentFlags().StringP(repoServerUserAccessSecretFlag, "k", "", "name of the secret having access credentials of the users that can connect to kopia repository server")
+	cmd.PersistentFlags().StringP(repoServerAdminUserAccessSecretFlag, "a", "", "name of the secret having admin credentials to connect to connect to kopia repository server")
+	cmd.PersistentFlags().StringP(repoPasswordSecretFlag, "r", "", "name of the secret containing password for the kopia repository")
+	cmd.PersistentFlags().StringP(prefixFlag, "p", "", "prefix to be set in kopia repository")
+	cmd.PersistentFlags().StringP(repoUserFlag, "z", "", "name of the user for accessing the kopia repository")
+	cmd.PersistentFlags().StringP(locationSecretFlag, "l", "", "name of the secret containing kopia repository storage location details")
+	cmd.PersistentFlags().StringP(locationCredsSecretFlag, "c", "", "name of the secret containing kopia repository storage credentials")
+
+	_ = cmd.MarkFlagRequired(tlsSecretFlag)
+	_ = cmd.MarkFlagRequired(repoServerUserFlag)
+	_ = cmd.MarkFlagRequired(repoServerUserAccessSecretFlag)
+	_ = cmd.MarkFlagRequired(repoServerAdminUserAccessSecretFlag)
+	_ = cmd.MarkFlagRequired(repoPasswordSecretFlag)
+	_ = cmd.MarkFlagRequired(prefixFlag)
+	_ = cmd.MarkFlagRequired(locationSecretFlag)
+	_ = cmd.MarkFlagRequired(locationCredsSecretFlag)
+	return cmd
+}
+
+func createNewRepositoryServer(cmd *cobra.Command, args []string) error {
+	cmd.SilenceUsage = true
+	if len(args) > 0 {
+		return newArgsLengthError("expected 0 args. Got %#v", args)
+	}
+
+	rsParams, err := generateRepositoryServerParams(cmd)
+	if err != nil {
+		return err
+	}
+
+	repositoryServer, err := validateSecretsAndConstructRepositoryServer(rsParams)
+	if err != nil {
+		return err
+	}
+	_, crCli, _, err := initializeClients()
+	if err != nil {
+		return err
+	}
+	ctx := context.Background()
+	rs, err := crCli.CrV1alpha1().RepositoryServers(defaultKanisterNamespace).Create(ctx, repositoryServer, metav1.CreateOptions{})
+	if err != nil {
+		return err
+	}
+	fmt.Printf("repository-server '%s' created\n", rs.GetName())
+	return nil
+}
+
+func generateRepositoryServerParams(cmd *cobra.Command) (*repositoryServerParams, error) {
+	// Fetch values of the flags
+	tlsSecret, _ := cmd.Flags().GetString(tlsSecretFlag)
+	repositoryServerUser, _ := cmd.Flags().GetString(repoServerUserFlag)
+	repositoryServerUserAccessSecret, _ := cmd.Flags().GetString(repoServerUserAccessSecretFlag)
+	repositoryServerAdminUserAccessSecret, _ := cmd.Flags().GetString(repoServerAdminUserAccessSecretFlag)
+	repositoryUser, _ := cmd.Flags().GetString(repoUserFlag)
+	repositoryPassword, _ := cmd.Flags().GetString(repoPasswordSecretFlag)
+	prefix, _ := cmd.Flags().GetString(prefixFlag)
+	location, _ := cmd.Flags().GetString(locationSecretFlag)
+	locationCreds, _ := cmd.Flags().GetString(locationCredsSecretFlag)
+
+	if strings.Contains(tlsSecret, "/") {
```

Comment:
Should we add regex validation here instead of this check?

Reply:
I was thinking to keep the code signature similar as we're doing for other CRs in `kanctl`